vignettes/algorithm.Rmd

```diff
@@ -0,0 +1,341 @@
+---
+title: "Technical Documentation"
+author: "Johan Larsson"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+bibliography: "sgdnet.bib"
+vignette: >
+  %\VignetteIndexEntry{Technical Documentation}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Background
+
+**sgdnet** uses the incremental gradient method SAGA [@defazio2014], which
+is a modification of the Stochastic Average Gradient (SAG) algorithm
+introduced in
+@schmidt2017. SAGA handles both strongly and
+non-strongly convex objectives -- even in the composite case -- making it
+applicable to a wide range of problems such as generalized
+linear models with elastic net regularization [@zou2005], which is the problem
+that **sgdnet** is designed to handle.
+
+SAGA has a simple convergence proof and uses a step size that in fact
+adjusts automatically to the magnitude of strong convexity in the 
+objective. This makes it practical for the end user, who avoids having
+to tune step size by hand.
+
+The purpose of this vignette is to present the algorithm as it is 
+implemented in **sgdnet** and to serve as guidance for anyone who is interested
+contributing to the development of the package.
+
+## Model setup
+
+Before the algorithm is set loose on data, its parameters have to be set up 
+properly and its data (possibly) preprocessed. For illustrative purposes,
+we will proceed by example and use Gaussian univariate regression
+on the `trees` dataset. We will attempt to predict the volume of a tree based
+on girth and height.
+
+```{r}
+x <- with(trees, cbind(Girth, Height))
+y <- trees$Volume
+```
+
+### Standardization
+
+By default, the feature matrix in **sgdnet** is centered and scaled to unit
+variance. Here, however, we used the *biased* population standard deviation
+and divide by $n$ rather than $n-1$.
```

Comment:
I'd move this to a footnote and rephrase. We're using the *sample* standard deviation here since we are only concerned with fitting the model to the data at hand. Since we aren't concerned with a population, biasedness (or not) is irrelevant